packages/commerce-sdk-react/src/hooks/useQuery.ts

```diff
@@ -1,18 +1,42 @@
 /*
- * Copyright (c) 2022, Salesforce, Inc.
+ * Copyright (c) 2023, Salesforce, Inc.
  * All rights reserved.
  * SPDX-License-Identifier: BSD-3-Clause
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 import {useQuery as useReactQuery, UseQueryOptions, QueryKey} from '@tanstack/react-query'
-import useAuthenticatedClient from './useAuthenticatedClient'
-import {IQueryFunction} from './types'
+import {useAuthorizationHeader} from './useAuthorizationHeader'
+import {ApiClient, ApiMethod, ApiOptions, MergedOptions} from './types'
+import {hasAllKeys, mergeOptions} from './utils'
 
-export const useQuery = <TData>(
-    queryKey: QueryKey,
-    fn: IQueryFunction<TData>,
-    queryOptions?: UseQueryOptions<TData, Error>
+export const useQuery = <
+    Options extends Omit<ApiOptions, 'body'>,
+    Client extends ApiClient,
+    Data,
+    Err,
+    QK extends QueryKey
+>(
+    apiOptions: Options,
+    queryOptions: UseQueryOptions<Data, Err, Data, QK>,
+    hookConfig: {
+        client: Client
+        method: ApiMethod<Options, Data>
+        getQueryKey: (options: MergedOptions<Client, Options>) => QK
+        requiredParameters: ReadonlyArray<keyof Options['parameters']>
+        enabled?: boolean
+    }
 ) => {
-    const authenticatedFn = useAuthenticatedClient(fn)
-    return useReactQuery<TData, Error>(queryKey, authenticatedFn, queryOptions)
+    const netOptions = mergeOptions(hookConfig.client, apiOptions)
+    const authenticatedMethod = useAuthorizationHeader(hookConfig.method)
+    return useReactQuery(
+        // End user can override queryKey if they really want to...
+        queryOptions.queryKey ?? hookConfig.getQueryKey(netOptions),
+        () => authenticatedMethod(apiOptions),
+        {
+            enabled:
+                hookConfig.enabled !== false &&
+                hasAllKeys(netOptions.parameters, hookConfig.requiredParameters),
+            ...queryOptions
```

Comment:
The `useQuery` default for `enabled` is "all required parameters are set". Individual query hooks can provide _additional_ checks via `hookConfig`, where the required parameters are still checked. The end user of the hook can provide an _override_ via `queryOptions` that ignores required parameters.